File: Lab Week 1 Part 1/dart/bin/encapsulation.dart
class BankAccount {
  double _balance = 0;

  double get balance => _balance;

  set balance(double value) {
    if (value < 0) throw ArgumentError('Balance cannot be negative');
    _balance = value;
  }

  void deposit(double amount) {
    if (amount <= 0) throw ArgumentError('Amount must be > 0');
    _balance += amount;
  }

  void withdraw(double amount) {
    if (amount <= 0 || amount > _balance) {
      throw ArgumentError('Invalid withdraw amount');
    }
    _balance -= amount;
  }
}

void run() {
  var acc = BankAccount();
  acc.deposit(100);
  acc.withdraw(40);
  print("Balance = ${acc.balance}"); //mirip Java System.out.println
}

File: Lab Week 1 Part 1/java/Encapsulation.java
public class Encapsulation {
    static class BankAccount {
        private double balance = 0;

        public double getBalance() { return balance; }

        public void deposit(double amount) {
            if (amount > 0) balance += amount;
        }

        public void withdraw(double amount) {
            if (amount > 0 && amount <= balance) balance -= amount;
        }
    }

    public static void run() {
        BankAccount acc = new BankAccount();
        acc.deposit(100);
        acc.withdraw(40);
        System.out.println("Balance: " + acc.getBalance());
    }
}
